cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -263,6 +313,11 @@ public OperationSnapshot createCluster(String name, String imageVersion, Map<Str
         clusterConfig.addTags(targetTag);
       }
 
+      // if internal ip is prefered then create dataproc cluster without external ip for better security
+      if (!conf.isPreferExternalIP()) {
```

Comment:
this is not correct, it will create internal IP clusters in many instances where CDAP requires the external IP. This should be using the useInternalIP flag.

Reply:
fixed.